### PR TITLE
Feature/edit task variables

### LINKED ISSUE #2: Show alert explanation
As a dashboard user, I want to have a button for showing an explanation, why this alert was triggered.
Implement this button in alert card. When clicking the button, request the TICK corresponding to the alert from Kapacitor and show the rule to the user.

### LINKED ISSUE #12: Make task variables editable

### PR DESCRIPTION
**Pull request by hefowe**
Closes #12 
Closes #2

### CODE REVIEW

File: src/apis/KapacitorApi.ts [hunk 1]
```diff
@@ -15,6 +17,28 @@ export default class KapacitorApi {
...
     }
+
+    public static async taskVariables(taskName: string): Promise<any> {
+        const response = await fetch(`http://82.140.0.78:9092/kapacitor/v1/tasks/${taskName}`);
+        const topic = await response.json();
+        return topic.vars;
+    }
+
+    public static async updateTaskVariables(taskName: string, taskVariables: any): Promise<string> {
+        await axios.patch(`http://82.140.0.78:9092/kapacitor/v1/tasks/${taskName}`, {
```
Comment (friedow) [inline src/apis/KapacitorApi.ts]: No need to use axios for that, `fetch()` is perfectly capable of sending PATCH requests. Please refer to the [fetch documentation](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch#Supplying_request_options) and remove axios from the dependencies afterwards.

File: src/components/TaskVariablesModal.vue [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        <div class="modal-container" :style="{ 'background-color': backgroundColor }">
+
+          <div class="modal-header">
+              <h1 class="header-title">Settings</h1>
+          </div>
+
+          <div class="modal-body">
+            <slot v-for="taskVariableKey in Object.keys(taskVariables)" name="body">
+              <label :key="taskVariableKey + 'label'" :for="taskVariableKey + 'input'">{{ toFreeText(taskVariableKey) }}:</label>
+              <input v-if="taskVariables[taskVariableKey].type === 'string' || taskVariables[taskVariableKey].type === 'lambda'" :key="taskVariableKey + 'input'" v-model="taskVariables[taskVariableKey].value" type="text">
```
Comment (friedow) [inline src/components/TaskVariablesModal.vue]: please break long lines

File: src/components/TaskVariablesModal.vue [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+              <input v-if="taskVariables[taskVariableKey].type === 'float' || taskVariables[taskVariableKey].type === 'duration'" :key="taskVariableKey + 'input'" v-model.number="taskVariables[taskVariableKey].value" type="number">
```
Comment (friedow) [inline src/components/TaskVariablesModal.vue]: please break long lines

File: src/components/TaskVariablesModal.vue [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    this.$emit('close');
+
+    await KapacitorApi.updateTaskVariables(this.taskName, this.taskVariables);
+  }
+  // endregion
+}
+</script>
+
+<style lang="less">
+
```
Comment (friedow) [inline src/components/TaskVariablesModal.vue:93]: Is this your code or do we need to state a reference to someone?

**Review by friedow — CHANGES_REQUESTED** [review]

File: src/components/TaskVariablesModal.vue (hunk 4 above)
Comment (hefowe) [reply src/components/TaskVariablesModal.vue:93]: https://vuejs.org/v2/examples/modal.html

**Comment by hefowe** [comment]
@friedow Please take another look

**Review by friedow — APPROVED** [review]

**Pull request merged**